src/v/security/authorizer.h

```diff
@@ -241,14 +250,23 @@ class authorizer final {
      * Compute whether the specified operation is allowed based on the implied
      * operations.
      */
-    bool acl_any_implied_ops_allowed(
+    std::optional<acl_matches::acl_match> acl_any_implied_ops_allowed(
       const acl_matches& acls,
       const acl_principal& principal,
       const acl_host& host,
       const acl_operation operation) const {
-        auto check_op = [&acls, &principal, &host](acl_operation operation) {
-            return acls.contains(
-              operation, principal, host, acl_permission::allow);
+        auto check_op = [&acls, &principal, &host]<typename T>(
+                          T begin,
+                          T end) -> std::optional<acl_matches::acl_match> {
```

Comment:
does this need to be a template. seems like you could have `[](auto begin, auto end)`?

Reply:
.... whoops